xdsl/dialects/arith.py

```diff
@@ -1223,6 +1227,64 @@ class MinnumfOp(FloatingPointLikeBinaryOperation):
     traits = traits_def(Pure())
 
 
+@irdl_op_definition
+class BitcastOp(IRDLOperation):
+    name = "arith.bitcast"
+
+    input = operand_def(signlessIntegerLike | floatingPointLike)
+    result = result_def(signlessIntegerLike | floatingPointLike)
+
+    assembly_format = "$input attr-dict `:` type($input) `to` type($result)"
+
+    def __init__(self, in_arg: SSAValue | Operation, target_type: Attribute):
+        super().__init__(operands=[in_arg], result_types=[target_type])
+
+    def verify_(self) -> None:
+        # check if we have a ContainerType
```

Comment:
this comment doesn't add much and it's a bit confusing since we first check for shaped types